modules/plugin/mongodb/src/test/java/org/geotools/data/mongodb/MongoSchemaFileStoreTest.java

```diff
@@ -95,6 +99,86 @@ public void test_Constructor_URI() throws IOException {
         }
     }
 
+    @Test
+    public void test_Constructor_URL() throws Exception {
+        final String mockFileURL = "https://mock.url.com/stations.json";
+
+        MockHTTPClient mockHttpClient = new MockHTTPClient();
+        byte[] responseBytes =
+                IOUtils.toByteArray(getClass().getResourceAsStream("stations-schema.json"));
+        MockHttpResponse mockResponse = new MockHttpResponse(responseBytes, "text/json");
+        mockHttpClient.expectGet(mockFileURL, mockResponse);
+
+        MongoSchemaStore mss =
+                new MongoSchemaFileStore("mockshema", new URL(mockFileURL), mockHttpClient);
```

Comment:
This constructor seems like is not used on runtime, try to test code used in production runtime.

Reply:
ok I will move the tests to instantiate data store through params

Reply:
removed constructors and unit tests updated